 #                                                  STATISTICAL THINKING

## Introduction
The Millennium Development Goals were a set of 8 goals for 2015 that were defined by the United Nations to help improve living conditions and the conditions of our planet. Key indicators were defined for each of these goals, to see whether they were being met. We will have a look at some of the key indicators from Goal 7: Ensure environmental sustainability, namely carbon dioxide emissions, protected land and sea areas, and forests.

* We will explore the dataset provided and analyse the polution behaviours of the countries over the past years.

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats as stats
from scipy.stats.stats import pearsonr

import warnings
warnings.filterwarnings('ignore')


sns.set()

In [2]:
mdg_data = pd.read_csv("Datasets/mdg.csv")
mdg_data.head()

CountryCode      Country  SeriesCode MDG  \
0            4  Afghanistan         567   Y   
1            4  Afghanistan         749   Y   
2            4  Afghanistan         616   Y   
3            8      Albania         567   Y   
4            8      Albania         749   Y   

                                              Series    1990 Footnotes Type  \
0  Proportion of land area covered by forest, per...     2.1       NaN    E   
1  Carbon dioxide emissions (CO2), thousand metri...  2676.9       NaN    G   
2  Terrestrial and marine areas protected to tota...    0.36       NaN    C   
3  Proportion of land area covered by forest, per...    28.8       NaN    C   
4  Carbon dioxide emissions (CO2), thousand metri...    7488       NaN    G   

     1991 Footnotes.1  ... Type.26 2017 Footnotes.27 Type.27 2018  \
0                      ...                                          
1  2493.6         NaN  ...                                          
2                      ...                                          
3                      ...                                          
4  3971.4         NaN  ...                                          

  Footnotes.28 Type.28 2019 Footnotes.29 Type.29  
0                                                 
1                                                 
2                                                 
3                                                 
4                                                 

[5 rows x 95 columns]

In [3]:
mdg_data.shape

(690, 95)

 The Dataframe has 690 rows and 95 Columns

#### Inspecting the data

In [4]:
#filling the empty spaces with NAN
mdg_data.replace(r'^\s*$', np.nan, regex=True, inplace = True)

###### checking and separating the dataset by series 


In [5]:
#number of countries represented
country = mdg_data['Country'].nunique()
print('number of countries =', country)

#number of series
series = mdg_data['Series'].nunique()
print('number of series = ',series)

series_list = mdg_data['Series'].unique()
print('series list = ', series_list)

number of countries = 230
number of series =  3
series list =  ['Proportion of land area covered by forest, percentage'
 'Carbon dioxide emissions (CO2), thousand metric tons of CO2 (CDIAC)'
 'Terrestrial and marine areas protected to total territorial area, percentage']


In [6]:
#seperating datasets based on series  

# Propotion of land area covered by forest DataFrame
mdg_proportion_of_land = mdg_data[mdg_data['Series'] ==  "Proportion of land area covered by forest, percentage"]
display('Proportion of land dataframe' ,mdg_proportion_of_land.head())

# Terrestrial and marine areas protected to total territorial area DataFrame
mdg_terrestrail = mdg_data[mdg_data['Series'] ==  'Terrestrial and marine areas protected to total territorial area, percentage']
display('Terrestrial and marine DataFrame' ,mdg_terrestrail.head())

# Carbon dioxide emissions (CO2) Dataframe
mdg_carbon_emission = mdg_data[mdg_data['Series'] ==  'Carbon dioxide emissions (CO2), thousand metric tons of CO2 (CDIAC)']
display('Carbon Emission DataFrame', mdg_carbon_emission.head())

'Proportion of land dataframe'

CountryCode         Country  SeriesCode MDG  \
0             4     Afghanistan         567   Y   
3             8         Albania         567   Y   
6            12         Algeria         567   Y   
9            16  American Samoa         567   Y   
12           20         Andorra         567   Y   

                                               Series  1990 Footnotes Type  \
0   Proportion of land area covered by forest, per...   2.1       NaN    E   
3   Proportion of land area covered by forest, per...  28.8       NaN    C   
6   Proportion of land area covered by forest, per...   0.7       NaN    C   
9   Proportion of land area covered by forest, per...    90       NaN    C   
12  Proportion of land area covered by forest, per...  35.6       NaN    E   

   1991  Footnotes.1  ... Type.26 2017  Footnotes.27 Type.27 2018  \
0   NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   
3   NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   
6   NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   
9   NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   
12  NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   

    Footnotes.28 Type.28 2019  Footnotes.29 Type.29  
0            NaN     NaN  NaN           NaN     NaN  
3            NaN     NaN  NaN           NaN     NaN  
6            NaN     NaN  NaN           NaN     NaN  
9            NaN     NaN  NaN           NaN     NaN  
12           NaN     NaN  NaN           NaN     NaN  

[5 rows x 95 columns]

'Terrestrial and marine DataFrame'

CountryCode         Country  SeriesCode MDG  \
2             4     Afghanistan         616   Y   
5             8         Albania         616   Y   
8            12         Algeria         616   Y   
11           16  American Samoa         616   Y   
14           20         Andorra         616   Y   

                                               Series  1990 Footnotes Type  \
2   Terrestrial and marine areas protected to tota...  0.36       NaN    C   
5   Terrestrial and marine areas protected to tota...  0.15       NaN    C   
8   Terrestrial and marine areas protected to tota...  5.97       NaN    C   
11  Terrestrial and marine areas protected to tota...  0.06       NaN    C   
14  Terrestrial and marine areas protected to tota...  5.53       NaN    C   

   1991  Footnotes.1  ... Type.26 2017  Footnotes.27 Type.27 2018  \
2   NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   
5   NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   
8   NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   
11  NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   
14  NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   

    Footnotes.28 Type.28 2019  Footnotes.29 Type.29  
2            NaN     NaN  NaN           NaN     NaN  
5            NaN     NaN  NaN           NaN     NaN  
8            NaN     NaN  NaN           NaN     NaN  
11           NaN     NaN  NaN           NaN     NaN  
14           NaN     NaN  NaN           NaN     NaN  

[5 rows x 95 columns]

'Carbon Emission DataFrame'

CountryCode         Country  SeriesCode MDG  \
1             4     Afghanistan         749   Y   
4             8         Albania         749   Y   
7            12         Algeria         749   Y   
10           16  American Samoa         749   Y   
13           20         Andorra         749   Y   

                                               Series     1990 Footnotes Type  \
1   Carbon dioxide emissions (CO2), thousand metri...   2676.9       NaN    G   
4   Carbon dioxide emissions (CO2), thousand metri...     7488       NaN    G   
7   Carbon dioxide emissions (CO2), thousand metri...  78924.8       NaN    G   
10  Carbon dioxide emissions (CO2), thousand metri...      NaN       NaN  NaN   
13  Carbon dioxide emissions (CO2), thousand metri...      NaN       NaN  NaN   

       1991  Footnotes.1  ... Type.26 2017  Footnotes.27 Type.27 2018  \
1    2493.6          NaN  ...     NaN  NaN           NaN     NaN  NaN   
4    3971.4          NaN  ...     NaN  NaN           NaN     NaN  NaN   
7   81220.4          NaN  ...     NaN  NaN           NaN     NaN  NaN   
10      NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   
13      NaN          NaN  ...     NaN  NaN           NaN     NaN  NaN   

    Footnotes.28 Type.28 2019  Footnotes.29 Type.29  
1            NaN     NaN  NaN           NaN     NaN  
4            NaN     NaN  NaN           NaN     NaN  
7            NaN     NaN  NaN           NaN     NaN  
10           NaN     NaN  NaN           NaN     NaN  
13           NaN     NaN  NaN           NaN     NaN  

[5 rows x 95 columns]

 #### MISSING VALUES AND EMPTY COLUMNS

* **Propotion of land area covered by forest**

In [7]:
mdg_proportion_of_land = mdg_proportion_of_land.dropna(how='all', axis='columns')
display('Proportion of Land  ' , mdg_proportion_of_land.head())


'Proportion of Land  '

CountryCode         Country  SeriesCode MDG  \
0             4     Afghanistan         567   Y   
3             8         Albania         567   Y   
6            12         Algeria         567   Y   
9            16  American Samoa         567   Y   
12           20         Andorra         567   Y   

                                               Series  1990 Footnotes Type  \
0   Proportion of land area covered by forest, per...   2.1       NaN    E   
3   Proportion of land area covered by forest, per...  28.8       NaN    C   
6   Proportion of land area covered by forest, per...   0.7       NaN    C   
9   Proportion of land area covered by forest, per...    90       NaN    C   
12  Proportion of land area covered by forest, per...  35.6       NaN    E   

    2000 Footnotes.10 Type.10  2005 Footnotes.15 Type.15  2010 Footnotes.20  \
0    2.1          NaN       E   2.1          NaN       E   2.1          NaN   
3   28.1          NaN       C  28.5          NaN       C  28.3          NaN   
6    0.7          NaN       C   0.6          NaN       C   0.6          NaN   
9     90          NaN       C    90          NaN       C    90          NaN   
12  35.6          NaN       E  35.6          NaN       E  35.6          NaN   

   Type.20  
0        E  
3        C  
6        C  
9        C  
12       E

* **Terrestrial and marine areas protected to total territorial**

In [8]:
mdg_terrestrail = mdg_terrestrail.dropna(how='all', axis='columns')
display('Terrestrail and Marine', mdg_terrestrail.head())



'Terrestrail and Marine'

CountryCode         Country  SeriesCode MDG  \
2             4     Afghanistan         616   Y   
5             8         Albania         616   Y   
8            12         Algeria         616   Y   
11           16  American Samoa         616   Y   
14           20         Andorra         616   Y   

                                               Series  1990 Type  2000  \
2   Terrestrial and marine areas protected to tota...  0.36    C  0.36   
5   Terrestrial and marine areas protected to tota...  0.15    C  0.48   
8   Terrestrial and marine areas protected to tota...  5.97    C  5.98   
11  Terrestrial and marine areas protected to tota...  0.06    C  0.06   
14  Terrestrial and marine areas protected to tota...  5.53    C  9.25   

   Type.10  2014 Type.24  
2        C  0.46       C  
5        C  1.91       C  
8        C  7.46       C  
11       C  8.62       C  
14       C  19.5       C

* **Carbon Emission**

In [9]:
mdg_carbon_emission = mdg_carbon_emission.dropna(how='all', axis='columns')
display('Carbon Emission', mdg_carbon_emission.head())

'Carbon Emission'

CountryCode         Country  SeriesCode MDG  \
1             4     Afghanistan         749   Y   
4             8         Albania         749   Y   
7            12         Algeria         749   Y   
10           16  American Samoa         749   Y   
13           20         Andorra         749   Y   

                                               Series     1990 Type     1991  \
1   Carbon dioxide emissions (CO2), thousand metri...   2676.9    G   2493.6   
4   Carbon dioxide emissions (CO2), thousand metri...     7488    G   3971.4   
7   Carbon dioxide emissions (CO2), thousand metri...  78924.8    G  81220.4   
10  Carbon dioxide emissions (CO2), thousand metri...      NaN  NaN      NaN   
13  Carbon dioxide emissions (CO2), thousand metri...      NaN  NaN      NaN   

   Type.1     1992  ...      2007 Type.17      2008 Type.18    2009 Type.19  \
1       G   1426.5  ...    2280.9       G    4217.1       G  6776.6       G   
4       G   2387.2  ...    4477.4       G    4657.1       G  4488.4       G   
7       G  81935.4  ...  109294.9       G  111575.8       G  121374       G   
10    NaN      NaN  ...       NaN     NaN       NaN     NaN     NaN     NaN   
13    NaN      NaN  ...       539       G       539       G     517       G   

        2010 Type.20      2011 Type.21  
1     8470.8       G   12251.4       G  
4     4415.1       G    4668.1       G  
7   119276.5       G  121755.4       G  
10       NaN     NaN       NaN     NaN  
13       517       G     491.4       G  

[5 rows x 49 columns]

#### DROPPING UNNECESARY COLUMNS

* In **propotion of Land covered by forest**, the following columns will remain : **Country, 1990, 2000, 2005, 2010**

* In **Terrestrail and Marine**, the following columns will remain: **Country, 1990, 2000, 2014**

* In **Carbon emission**, the following columns will remain: **Country, 1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011**

* **Propotion of Land covered by forest**

In [10]:
new_mdg_proportion_of_land = mdg_proportion_of_land.loc[:,('Country','1990', '2000', '2010')]
new_mdg_proportion_of_land.head()

Country  1990  2000  2010
0      Afghanistan   2.1   2.1   2.1
3          Albania  28.8  28.1  28.3
6          Algeria   0.7   0.7   0.6
9   American Samoa    90    90    90
12         Andorra  35.6  35.6  35.6

* **Terrestrial and marine areas protected to total territorial area**

In [11]:
new_mdg_terrestrail = mdg_terrestrail.loc[:,('Country','1990', '2000', '2014')]
new_mdg_terrestrail.head()

Country  1990  2000  2014
2      Afghanistan  0.36  0.36  0.46
5          Albania  0.15  0.48  1.91
8          Algeria  5.97  5.98  7.46
11  American Samoa  0.06  0.06  8.62
14         Andorra  5.53  9.25  19.5

* **Carbon dioxide emissions (CO2)**

In [12]:
new_mdg_carbon_emission  = mdg_carbon_emission.loc[:,('Country', '1990','1991','1992','1993','1994','1995','1996','1997','1998','1999','2000','2001','2002','2003','2004','2005','2006','2007','2008','2009','2010','2011')]
new_mdg_carbon_emission.head()

Country     1990     1991     1992     1993     1994     1995  \
1      Afghanistan   2676.9   2493.6   1426.5   1375.1   1320.1   1268.8   
4          Albania     7488   3971.4   2387.2   2343.2   1928.8   2086.5   
7          Algeria  78924.8  81220.4  81935.4  82294.8  86500.9  95345.7   
10  American Samoa      NaN      NaN      NaN      NaN      NaN      NaN   
13         Andorra      NaN      NaN      NaN      NaN      NaN      407   

       1996    1997      1998  ...     2002     2003     2004      2005  \
1    1199.1  1114.8    1056.1  ...    894.7   1037.8    957.1    1338.5   
4    2016.9  1543.8    1752.8  ...   3751.3   4294.1   4165.7    4253.7   
7   97131.5   88195  107080.1  ...  90853.6  92533.1  89493.1  107127.7   
10      NaN     NaN       NaN  ...      NaN      NaN      NaN       NaN   
13    425.4   458.4       484  ...    531.7    535.4    564.7     575.7   

        2006      2007      2008    2009      2010      2011  
1     1657.5    2280.9    4217.1  6776.6    8470.8   12251.4  
4       3865    4477.4    4657.1  4488.4    4415.1    4668.1  
7   100919.5  109294.9  111575.8  121374  119276.5  121755.4  
10       NaN       NaN       NaN     NaN       NaN       NaN  
13     546.4       539       539     517       517     491.4  

[5 rows x 23 columns]

#### Top 15 Missing values by country in descending order

In [13]:
#propotion of land area covered by forest
display(new_mdg_proportion_of_land.groupby('Country').apply(lambda x: x.isnull().sum().sum()).sort_values(ascending=False).head(15))

Country
China, Macao Special Administrative Region        3
Sudan                                             3
Monaco                                            3
Serbia and Montenegro                             3
South Sudan                                       3
China, Hong Kong Special Administrative Region    3
Yugoslavia [former Socialist Federal Republic]    3
Ghana                                             0
Germany                                           0
France                                            0
Gambia                                            0
Gibraltar                                         0
Gabon                                             0
Greece                                            0
French Polynesia                                  0
dtype: int64

In [14]:
# Area covered by terrestrail and marine
display(new_mdg_terrestrail.groupby('Country').apply(lambda x: x.isnull().sum().sum()).sort_values(ascending=False).head(15))

Country
Saint Helena                                      3
Yugoslavia [former Socialist Federal Republic]    3
Gibraltar                                         3
Wallis and Futuna Islands                         3
Saint Pierre and Miquelon                         3
Sudan (former)                                    3
Serbia and Montenegro                             3
Gabon                                             0
France                                            0
French Guiana                                     0
French Polynesia                                  0
Germany                                           0
Gambia                                            0
Georgia                                           0
Fiji                                              0
dtype: int64

In [15]:
# Carbon emission
display(new_mdg_carbon_emission.groupby('Country').apply(lambda x: x.isnull().sum().sum()).sort_values(ascending=False).head(15))

Country
Sudan                                             22
Guam                                              22
Mayotte                                           22
Tuvalu                                            22
Monaco                                            22
Puerto Rico                                       22
Tokelau                                           22
United States Virgin Islands                      22
South Sudan                                       22
San Marino                                        22
American Samoa                                    22
Northern Mariana Islands                          22
Western Sahara                                    22
Yugoslavia [former Socialist Federal Republic]    20
Liechtenstein                                     17
dtype: int64

* **Dropping countries with all null values and fill remain rows with null with mean** 

#### * PROPOTION OF LAND COVERED BY FOREST

In [16]:
#removing projects that have rows with all null or missing values
new_mdg_proportion_of_land = new_mdg_proportion_of_land[new_mdg_proportion_of_land.isnull().sum(axis=1) != 3]

#filling null values with mean
cols = new_mdg_proportion_of_land.columns.drop('Country')
new_mdg_proportion_of_land[cols] = new_mdg_proportion_of_land[cols].apply(pd.to_numeric, errors='coerce')

#function to replace NaN with row average
m = new_mdg_proportion_of_land.mean(axis=1)
for i, col in enumerate(new_mdg_proportion_of_land):
             # using i allows for duplicate columns
             # inplace *may* not always work here, so IMO the next line is preferred
             # df.iloc[:, i].fillna(m, inplace=True)
    new_mdg_proportion_of_land.iloc[:, i] = new_mdg_proportion_of_land.iloc[:, i].fillna(m)

new_mdg_proportion_of_land.head()

Country  1990  2000  2010
0      Afghanistan   2.1   2.1   2.1
3          Albania  28.8  28.1  28.3
6          Algeria   0.7   0.7   0.6
9   American Samoa  90.0  90.0  90.0
12         Andorra  35.6  35.6  35.6

#### TERRESTRAIL and MARINE

In [17]:
#Rows or countries with missing value less than 3 will remain, removing rows with all missing data
new_mdg_terrestrail = new_mdg_terrestrail[new_mdg_terrestrail.isnull().sum(axis=1) != 3]

#preparing columns to be converted from object to numeric
cols = new_mdg_terrestrail.columns.drop('Country')

#converting columns into numeric
new_mdg_terrestrail[cols] = new_mdg_terrestrail[cols].apply(pd.to_numeric, errors='coerce')

#function to replace NaN with row average
m = new_mdg_terrestrail.mean(axis=1)
for i, col in enumerate(new_mdg_terrestrail):
             # using i allows for duplicate columns
             # inplace *may* not always work here, so IMO the next line is preferred
             # df.iloc[:, i].fillna(m, inplace=True)
    new_mdg_terrestrail.iloc[:, i] = new_mdg_terrestrail.iloc[:, i].fillna(m)

new_mdg_terrestrail.head()

Country  1990  2000   2014
2      Afghanistan  0.36  0.36   0.46
5          Albania  0.15  0.48   1.91
8          Algeria  5.97  5.98   7.46
11  American Samoa  0.06  0.06   8.62
14         Andorra  5.53  9.25  19.50

 #### CARBON  EMISSION

In [18]:
# dropping rows with all missing value or no data
new_mdg_carbon_emission = new_mdg_carbon_emission[new_mdg_carbon_emission.isnull().sum(axis=1) != 22]

# Selecting columns to be converted to numeric
cols = new_mdg_carbon_emission.columns.drop('Country')

#converting columns into numeric
new_mdg_carbon_emission[cols] = new_mdg_carbon_emission[cols].apply(pd.to_numeric, errors='coerce')

#function to replace NaN with row average
m = new_mdg_carbon_emission.mean(axis=1)
for i, col in enumerate(new_mdg_carbon_emission):
             # using i allows for duplicate columns
             # inplace *may* not always work here, so IMO the next line is preferred
             # df.iloc[:, i].fillna(m, inplace=True)
    new_mdg_carbon_emission.iloc[:, i] = new_mdg_carbon_emission.iloc[:, i].fillna(m)

new_mdg_carbon_emission.head()

Country          1990          1991          1992          1993  \
1   Afghanistan   2676.900000   2493.600000   1426.500000   1375.100000   
4       Albania   7488.000000   3971.400000   2387.200000   2343.200000   
7       Algeria  78924.800000  81220.400000  81935.400000  82294.800000   
13      Andorra    511.429412    511.429412    511.429412    511.429412   
16       Angola   4429.700000   4367.400000   4418.700000   5801.200000   

            1994     1995     1996     1997      1998  ...     2002     2003  \
1    1320.100000   1268.8   1199.1   1114.8    1056.1  ...    894.7   1037.8   
4    1928.800000   2086.5   2016.9   1543.8    1752.8  ...   3751.3   4294.1   
7   86500.900000  95345.7  97131.5  88195.0  107080.1  ...  90853.6  92533.1   
13    511.429412    407.0    425.4    458.4     484.0  ...    531.7    535.4   
16   3890.700000  11012.0  10491.3   7381.7    7308.3  ...  12665.8   9064.8   

       2004      2005      2006      2007      2008      2009      2010  \
1     957.1    1338.5    1657.5    2280.9    4217.1    6776.6    8470.8   
4    4165.7    4253.7    3865.0    4477.4    4657.1    4488.4    4415.1   
7   89493.1  107127.7  100919.5  109294.9  111575.8  121374.0  119276.5   
13    564.7     575.7     546.4     539.0     539.0     517.0     517.0   
16  18793.4   19156.4   22266.0   25152.0   27172.5   29361.7   29743.0   

        2011  
1    12251.4  
4     4668.1  
7   121755.4  
13     491.4  
16   29710.0  

[5 rows x 23 columns]

* Propotion of Land covered by forest data was collected in 1990, 2000, 2005 and 2010

* Terrestrail and Marine data was collected in 1990, 2000 and 2014

* Carbon emission data was collected from 1990 to 2011

**Which is why necessary to seperate the data based on the series**

# 1.Countries represented. Missing values are there by country, year and series

* **Countries represented in each series**

In [19]:
Land_proportion_countries = new_mdg_proportion_of_land.Country.nunique()
print('Countries represented in Proportion of land covered by forest =',Land_proportion_countries)

Terrestrail_marine_countries = new_mdg_terrestrail.Country.nunique()
print('Countries represented in Terrestrial and marine areas protected to total territorial =',Terrestrail_marine_countries)

Carbon_emission_countries = new_mdg_carbon_emission.Country.nunique()
print('Countries represented in Carbon dioxide emissions (CO2) =',Carbon_emission_countries)

Countries represented in Proportion of land covered by forest = 223
Countries represented in Terrestrial and marine areas protected to total territorial = 223
Countries represented in Carbon dioxide emissions (CO2) = 217


* **Missing values by country, year and series**

From the cleaned datasets there are no missing values

In [20]:
Land_proportion_missing = new_mdg_proportion_of_land.isnull().sum().sum()
print('Missing values in Proportion of land covered by forest =',Land_proportion_missing)

Terrestrail_marine_missing = new_mdg_terrestrail.isnull().sum().sum()
print('Missing values in Terrestrial and marine areas protected to total territorial =',Terrestrail_marine_missing)

Carbon_emission__missing = new_mdg_carbon_emission.isnull().sum().sum()
print('Missing values in Carbon dioxide emissions (CO2) =',Carbon_emission__missing)

Missing values in Proportion of land covered by forest = 0
Missing values in Terrestrial and marine areas protected to total territorial = 0
Missing values in Carbon dioxide emissions (CO2) = 0


# 2. Top and bottom 5 countries in terms of C02 emissions in 1990 and their emissions and how they changed by 2011

* **Top 5 countries in CO2 emmission in 1990 their changes in 2011**

In [27]:
Carbon_emission_top = new_mdg_carbon_emission.loc[:,("Country", "1990","2011")].sort_values(by = '1990', ascending=False)
Carbon_emission_top.head()    

Country          1990       2011
652       United States  4.823557e+06  5305569.6
124               China  2.460744e+06  9019518.2
499  Russian Federation  1.660534e+06  1808073.0
301               Japan  1.094288e+06  1187657.0
229             Germany  8.276009e+05   729458.0

* USA's Carbon emission increased by 9.99%, which is approximately **10%** 
* China's Carbon emission increased by **266.54%** 
* Russian federation increased by **8.89%**
* Japan increased by **8.53%**
* Germany has decreased by **11.86%**

* **Bottom 5 countries in CO2 emmission in 1990 their changes in 2011**

In [31]:
Carbon_emission_bottom = new_mdg_carbon_emission.loc[:,("Country", "1990","2011")].sort_values(by = '1990', ascending=True)
Carbon_emission_bottom.head()

Country    1990     2011
679         Yemen -2999.6  22295.4
445          Niue     3.7     11.0
505  Saint Helena     7.3     11.0
559       Somalia    18.3    575.7
142  Cook Islands    22.0     69.7

Yemen seem to have negative value, need to be changed to positive

In [34]:
#changing negative values in 1990 column
new_mdg_carbon_emission['1990'] = new_mdg_carbon_emission['1990'].abs()

In [35]:
Carbon_emission_bottom = new_mdg_carbon_emission.loc[:,("Country", "1990","2011")].sort_values(by = '1990', ascending=True)
Carbon_emission_bottom.head()

Country  1990   2011
445          Niue   3.7   11.0
505  Saint Helena   7.3   11.0
559       Somalia  18.3  575.7
313      Kiribati  22.0   62.3
142  Cook Islands  22.0   69.7

* Niue inceased by **7.3** thousand metric tons
* Saint Helena increased by **3.7** thousand metric tons
* Somalia increased by **557.4** thousand metric tons
* Kiribati increased by **40.3** thousand metric tons
* Cook Islands **47.7** thousand metric tons